server/chat-plugins/datasearch.js

```diff
@@ -925,20 +924,10 @@ function runMovesearch(target, cmd, canAll, message) {
 
 			let template = Dex.getTemplate(target);
 			if (template.exists) {
-				if (targetMons.includes(template.name)) return {reply: "A search cannot include the same Pok\u00e9mon twice."};
 				if (parameters.length > 1) return {reply: "A Pok\u00e9mon learnset cannot have alternative parameters."};
-				if (!template.learnset) template = Dex.getTemplate(template.baseSpecies);
-				lsetData[template.name] = Object.assign({}, template.learnset);
-				targetMons.push(template.name);
-
-				let originalTemplateName = template.name;
-				while (template.prevo) {
-					template = Dex.getTemplate(template.prevo);
-					for (let move in template.learnset) {
-						if (!lsetData[originalTemplateName][move]) lsetData[originalTemplateName][move] = template.learnset[move];
-					}
-				}
-				orGroup.skip = true;
+				if (targetMons.some(mon => mon.name === template.name && isNotSearch !== mon.shouldBeExcluded)) return {reply: "A search cannot both exclude and include the same Pok\u00e9mon."};
```

Comment:
How large is `targetMons` here? Doing basically the same iteration over the list twice is kind of ugh. Isn't this just:
```ts
for (const mon of targetMons) {
  if (mon.name === template.name) {
    if (isNotSearch !== mon.shouldBeExcluded) return ...
    return ...
  }
}
```

Reply:
targetMons is only as big as the number of pokemon that the user enters, so practically 5-6 at most I would say.